v2/pkg/genruntime/resolved_secrets.go

```diff
@@ -30,3 +30,14 @@ func (r ResolvedSecrets) LookupSecret(ref SecretReference) (string, error) {
 	}
 	return result, nil
 }
+
+// MustLookupSecret looks up the secret value for the given reference. If the reference is nil, an error is returned.
+// If the secret cannot be found, an error is returned
+// TODO: Wondering if there's a better name for this than what I've got?
```

Comment:
Remove this note... but also if anybody has a better name feel free to suggest it

Reply:
Why is this not just `LookupSecret()`?

I thought the Go ecosystem convention for the prefix `Must` was that such methods would panic, and not return errors.

Given that this one _can_ fail, maybe `TryLookupSecret()`?

Reply:
Yes you're right, `Must` is not the right prefix. The reason I'm not using `LookupSecret` is that already exists. The difference is that this ones signature accepts a ptr `*SecretReference`.

Reply:
Maybe just `LookupSecretPtr` (even though it's a bit horrible it's clearer than `Must`)

Reply:
Have gone with `LookupSecretPtr` - let me know if you think there's a better name.

Reply:
Ah, my kingdom for some method overloads. 😁

Reply:
Will do `LookupSecretFromPtr` after discussion for extra clarity.